istioctl/cmd/proxyconfig.go

```diff
@@ -190,6 +303,94 @@ func proxyConfig() *cobra.Command {
 	listenerConfigCmd.PersistentFlags().StringVar(&listenerType, "type", "", "Filter listeners by type field")
 	listenerConfigCmd.PersistentFlags().IntVar(&port, "port", 0, "Filter listeners by Port field")
 
+	loggingCmd := &cobra.Command{
+		Use:   "logging <pod-name[.namespace]>",
+		Short: "(experimental) Retrieves logging levels of the Envoy in the specified pod",
+		Long:  "(experimental) Retrieve information about logging levels of the Envoy instance in the specified pod, and update optionally",
+		Example: `  # Retrieve information about logging levels for a given pod from Envoy.
+  istioctl proxy-config logging <pod-name[.namespace]>
+
+  # Update levels of the specified loggers and retrieve all the information about logging levels.
+  istioctl proxy-config logging <pod-name[.namespace]> --log_level=all:debug
+
+  # Reset levels of all the loggers to default value (warning) and retrieve all the information about logging levels.
+  istioctl proxy-config logging <pod-name[.namespace]> -r
+`,
+		Aliases: []string{"o"},
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				cmd.Println(cmd.UsageString())
+				return fmt.Errorf("logging requires pod name")
+			}
+			return nil
+		},
+		RunE: func(c *cobra.Command, args []string) error {
+			destLoggerLevels := map[string]Level{}
+			if reset {
+				// reset logging level to `defaultOutputLevel`, and ignore the `--log_level` option
```

Comment:
`--log_level` isn't a great UX -- I think normally we do `-` instead of `_` (maybe not in istioctl, but in other tools). Can we just make this `--level` or `--set`?

Reply:
`log` is already in the command. No need to repeat IMO. `--level` seems okay.

`istioctl x pc log <pod>.<namespace> --level connection:info`

Alternatively, do we need a flag at all?

- no args == print the current levels
- args after pod name == logger:level pairs

Reply:
I prefer using the flag, as following idioms of `istioctl proxy-config` command.